core/commonMain/src/kotlinx/serialization/modules/SerializersModuleCollector.kt

```diff
@@ -24,7 +24,14 @@ public interface SerializersModuleCollector {
     /**
      * Accept a serializer, associated with [kClass] for contextual serialization.
      */
-    public fun <T : Any> contextual(kClass: KClass<T>, serializer: KSerializer<T>)
+    public fun <T : Any> contextual(kClass: KClass<T>, serializer: KSerializer<T>): Unit =
+        contextual(kClass) { _ -> serializer }
```

Comment:
why default implementation is needed here, as looks like it will be always overridden?

Reply:
Nope, it is not overriden, for example, in PolymorphismValidator (since it does not use contextual at all)